python/paddle/fluid/layers/learning_rate_scheduler.py

```diff
@@ -106,14 +107,14 @@ def exponential_decay(learning_rate, decay_steps, decay_rate, staircase=False):
           sgd_optimizer.minimize(avg_cost)
 
     """
-    global_step = _decay_step_counter()
-
-    with init_on_cpu():
-        # update learning_rate
-        div_res = global_step / decay_steps
-        if staircase:
-            div_res = ops.floor(div_res)
-        decayed_lr = learning_rate * (decay_rate**div_res)
+    with default_main_program().pre_optimize_guard():
+        global_step = _decay_step_counter()
+        with init_on_cpu():
+            # update learning_rate
+            div_res = global_step / decay_steps
```

Comment:
I haven't tested this, does it works well on `elementwise_xxx` op which be patched here: https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/fluid/layers/math_op_patch.py#L139

Reply:
The `test_dist_transpiler.py` will test most of the cases.